packages/mui-base/src/Switch/Switch.types.ts

```diff
@@ -74,28 +78,28 @@ export type SwitchOwnerState = Simplify<
   }
 >;
 
-export type SwitchRootSlotProps = {
+export type SwitchRootSlotProps = Simplify<
+  UseSwitchRootSlotProps & {
+    ownerState: SwitchOwnerState;
+    className?: string;
+    children?: React.ReactNode;
+  }
+>;
+
+export type SwitchThumbSlotProps = {
   ownerState: SwitchOwnerState;
   className?: string;
   children?: React.ReactNode;
 };
 
-export type SwitchThumbSlotProps = {
+export type SwitchTrackSlotProps = {
   ownerState: SwitchOwnerState;
   className?: string;
   children?: React.ReactNode;
 };
 
-export type SwitchTrackSlotProps = {
+export type SwitchInputSlotProps = {
```

Comment:
This should include the `UseSwitchInputSlotProps` like before. Was it causing any problems?

Reply:
Now it doesn't cause any problem. I think I had the wrong idea that inputSlot shouldn't get any props.